Mediapipe.Net/Framework/Format/ImageFrame.cs

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) homuler and Vignette
+// This file is part of MediaPipe.NET.
+// MediaPipe.NET is licensed under the MIT License. See LICENSE for details.
+
+using System;
+using Mediapipe.Net.Core;
+using Mediapipe.Net.Native;
+
+namespace Mediapipe.Net.Framework.Format
+{
+    public class ImageFrame : MpResourceHandle
+    {
+        public static readonly uint DefaultAlignmentBoundary = 16;
+        public static readonly uint GlDefaultAlignmentBoundary = 4;
+
+        public delegate void Deleter(IntPtr ptr);
+
+        public ImageFrame() : base()
+        {
+            UnsafeNativeMethods.mp_ImageFrame__(out var ptr).Assert();
+            Ptr = ptr;
+        }
+
+        public ImageFrame(IntPtr imageFramePtr, bool isOwner = true) : base(imageFramePtr, isOwner) { }
+
+        public ImageFrame(ImageFormat format, int width, int height) : this(format, width, height, DefaultAlignmentBoundary) { }
+
+        public ImageFrame(ImageFormat format, int width, int height, uint alignmentBoundary) : base()
+        {
+            UnsafeNativeMethods.mp_ImageFrame__ui_i_i_ui(format, width, height, alignmentBoundary, out var ptr).Assert();
+            Ptr = ptr;
+        }
+
+        // NOTE: This byte* was a NativeArray<byte> from Unity. See if it adapts.
+        unsafe public ImageFrame(ImageFormat format, int width, int height, int widthStep, byte* pixelData)
+        {
+            unsafe
+            {
+                UnsafeNativeMethods.mp_ImageFrame__ui_i_i_i_Pui8_PF(
+                  format, width, height, widthStep,
+                  (IntPtr)pixelData,
+                  releasePixelData,
+                  out var ptr
+                ).Assert();
+                Ptr = ptr;
+            }
+        }
+
+        protected override void DeleteMpPtr() => UnsafeNativeMethods.mp_ImageFrame__delete(Ptr);
+
+        // [AOT.MonoPInvokeCallback(typeof(Deleter))] (?)
```

Comment:
This is a destructor so I assume this is a pointer that should be disposed?

Reply:
By pointer you mean the pixel data? idk.
Didn't know what `[AOT.MonoPInvokeCallback(typeof(Deleter))]`, so it's a destructor?

Reply:
that AOT callback tells that this specific P/Invoke call is a deleter or a destructor so I think we should reflect this as that.